flyMultiPlateScript.m

```diff
@@ -200,14 +208,42 @@ function flyMultiPlateScript()
     timestampFileName = strrep(timestampFileName,'.mp4','-timestamps.csv');
     timestampFileName = strrep(timestampFileName,'.m4v','-timestamps.csv');
 
-    %Check the existence of the associated timstamp file
     curFolder = pwd;
+
+    %Check the existence of the associated timstamp file
     cd(pathName);
     if not(exist(timestampFileName,'file') == 2)
         [timestampFileName,timestampPathName] = uigetfile({'*.csv'},strcat('Select the timestamp file associated with: ',fileName));
+        %Changing to this directory assuming we'll find the wellposes file there too
+        cd(timestampPathName);
     else
         timestampPathName = pathName;
     end
+
+    if useSavedWells == 1
+
+        %% Find the well positions file
+
+        % Assume the well positions file is in the same place as the timestamps file
+        wellposesFileName = strrep(timestampFileName,'-wellposes');
+        tmpwellposesFileName = strcat(wellposesFileName,'.mat');
+        wellposesPathName = timestampPathName;
+
+        %Check the existence of the associated wellposes file
+        if not(exist(tmpwellposesFileName,'file') == 2)
+            [tmpwellposesFileName,wellposesPathName] = uigetfile({'*.csv'},strcat('Select the timestamp file associated with: ',fileName));
+            %Remove the .mat for the load function (which doesn't take the filename with '.mat')
+            wellposesFileName = strrep(tmpwellposesFileName,'.mat','');
+        else
+            wellposesPathName = pathName;
+        end
+
+        %Load the wellposes mat file (two temp variables we'll use later)
+        cd(wellposesPathName);
+        load(wellposesFileName);
```

Comment:
I'd suggest something like `load(fullfile('wellposesPathName','wellposesFileName'))` to avoid using `cd`.